Description: fix various deprecations which became fatal in perl 5.26
 - escape { in regexps
 - replace POSIX::tmpnam() with File::Temp's backward compatibility equivalent
 - additionally drop assignment to $[
Origin: vendor
Bug-Debian: https://bugs.debian.org/866317
 https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=740782
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2017-07-15

--- a/html2ps
+++ b/html2ps
@@ -355,6 +355,7 @@
 
 use POSIX;
 $posix = 1;
+use File::Temp qw/ :POSIX /;
 
 %extend=('quote',1, 'font',1, 'colour',1, 'hyphenation',1);
 %fal=("serif","times", "sans_serif","helvetica", "monospace","courier");
@@ -494,7 +495,7 @@
  if($opt_D && !$package{'Ghostscript'});
 die "Ghostscript is required to generate cross references\n"
  if($opt_R && !$package{'Ghostscript'});
-$tmpname=$posix?POSIX::tmpnam():"h2p_$$";
+$tmpname=$posix?tmpnam():"h2p_$$";
 sysopen TMP, $tmpname, O_RDWR|O_CREAT|O_EXCL, 0600 or die "$!";
 close TMP;
 ($scr=$tmpname)=~/\w+$/;
@@ -3406,7 +3407,6 @@
   local($optlist)=@_;
   local(@args,$_,$opt,$opts,$rest,$olist,$plist,$found,@popts);
   local($errs)=0;
-  local($[)=0;
   @args=split( /\|/, $optlist );
   for $opt (@args) {
     if(substr($opt,-1,1) ne ':') {$olist.=$opt}
@@ -3831,7 +3831,7 @@
       $temp=$'=~/\/P\d+_?\d* \{/?$`:$rest;
       ($eps{$pid})=$temp=~/([\w\W]*)} D/;
     }
-    if(/\/DH {/) {
+    if(/\/DH \{/) {
       $'=~/%EndDH/;
       $ph="/DH {$`";
     }
@@ -4082,7 +4082,7 @@
   }
   if(open(HYPH,"<$hyfile")) {
     &dbg("Reading hyphenation patterns from $hyfile\n") if($opt_d);
-    <HYPH>=~/\\patterns{.*/;
+    <HYPH>=~/\\patterns\{.*/;
     close HYPH;
     $def=$`;
     ($patterns=$')=~s/\^\^($X$X)/chr hex $1/eg;
